src/components/cursor-controller.js

```diff
@@ -89,161 +83,68 @@ AFRAME.registerComponent("cursor-controller", {
-      this.data.camera.object3D.getWorldPosition(cameraPos);
-      cameraPos.y = this.data.cursor.object3D.position.y;
-      this.data.cursor.object3D.lookAt(cameraPos);
     };
   })(),
 
-  updateDistanceAndTargetType: function() {
-    const intersection = this.intersection;
-    if (intersection && intersection.distance <= this.data.far) {
-      this.data.cursor.object3D.position.copy(intersection.point);
-      this.currentDistance = intersection.distance;
-    } else {
-      this.currentDistance = this.data.far;
-      this.data.cursor.object3D.position.copy(this.raycaster.ray.origin);
-      this.data.cursor.object3D.position.addScaledVector(this.raycaster.ray.direction, this.currentDistance);
-    }
-
-    if (!intersection) {
-      this.currentTargetType = TARGET_TYPE_NONE;
-    } else if (intersection.object.el.matches(".interactable, .interactable *")) {
-      this.currentTargetType = TARGET_TYPE_INTERACTABLE;
-    } else if (intersection.object.el.matches(".ui, .ui *")) {
-      this.currentTargetType = TARGET_TYPE_UI;
-    }
-  },
-
-  _isTargetOfType: function(mask) {
-    return (this.currentTargetType & mask) === this.currentTargetType;
-  },
-
-  setCursorVisibility: function(visible) {
-    this.data.cursor.setAttribute("visible", visible);
-    this.el.setAttribute("line", { visible: visible && this.data.drawLine });
-  },
-
-  forceCursorUpdate: function() {
-    this.performRaycast(this.targets);
-    this.updateDistanceAndTargetType();
-    this.data.cursor.components["static-body"].syncToPhysics();
-  },
-
-  isInteracting: function() {
-    return this.data.cursor.components["super-hands"].state.has("grab-start");
-  },
-
-  startInteraction: function() {
-    if (this._isTargetOfType(TARGET_TYPE_INTERACTABLE_OR_UI)) {
-      this.data.cursor.emit("cursor-grab", {});
-      return true;
-    }
-    return false;
-  },
-
-  endInteraction: function() {
-    this.data.cursor.emit("cursor-release", {});
-  },
-
-  moveCursor: function(x, y) {
-    this.mousePos.set(x, y);
-  },
-
-  changeDistanceMod: function(delta) {
-    const { near, far } = this.data;
-    const targetDistanceMod = this.currentDistanceMod + delta;
-    const moddedDistance = this.currentDistance - targetDistanceMod;
-    if (moddedDistance > far || moddedDistance < near) {
-      return false;
-    }
-
-    this.currentDistanceMod = targetDistanceMod;
-    return true;
-  },
-
-  _handleCursorLoaded: function() {
-    this.data.cursor.object3DMap.mesh.renderOrder = window.APP.RENDER_ORDER.CURSOR;
-    this.data.cursor.removeEventListener("loaded", this._handleCursorLoaded);
-  },
-
   remove: function() {
-    this.emitIntersectionEvents(this.intersection, null);
-    this.intersection = null;
-    this.data.cursor.removeEventListener("loaded", this._handleCursorLoaded);
+    this.emitIntersectionEvents(this.prevIntersection, null);
+    delete this.prevIntersection;
   }
 });
```

Comment:
I remember when this file was longer than this and also depended on a 500-line A-Frame component.

![eagle](https://user-images.githubusercontent.com/428637/47402044-8ff17680-d6f8-11e8-8493-4a593c7af797.jpg)